earth2mip/beta/perturbation/brown.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+
+import numpy as np
+import torch
+
+
+class Brown:
+    """Lat/Lon 2D brown noise
+
+    Parameters
+    ----------
+    noise_amplitude : float, optional
+        Noise amplitude, by default 0.05
+    reddening : int, optional
+        Reddening in Fourier space, by default 2
+    """
+
+    def __init__(self, noise_amplitude: float = 0.05, reddening: int = 2):
+        self.reddening = reddening
+        self.noise_amplitude = noise_amplitude
+
+    @torch.inference_mode()
+    def __call__(
+        self,
+        x: torch.Tensor,
+        coords: OrderedDict[str, np.ndarray],
+    ) -> torch.Tensor:
+        """Apply perturbation method
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor intended to apply perturbation on
+        coords : OrderedDict[str, np.ndarray]
+            Ordered dict representing coordinate system that discribes the tensor, must
+            contain "lat" and "lon" coordinates
+
+        Returns
+        -------
+        torch.Tensor
+            Perturbation noise tensor
+        """
+        shape = x.shape
+        dims = list(coords.keys())
+        # Can be generalized if needed to have coords a constructor parameter
+        if "lat" not in dims or "lon" not in dims:
+            raise ValueError("Input tensor coords needs to contian lat and lon dims")
```

Comment:
For the sake of reusability, we should construct a generic utility function that can do this sort of handshaking.

Maybe it looks like this:
```python
def handshake_coords(input_coords: OrderedDict, required_coords: OrderedDict) -> None:
    """ Tests whether input_coords and required_coords are compatible. """
```

Reply:
Good call 100% agree